src/Umbraco.Core/Exceptions/ResolveUnregisteredDependencyException.cs

```diff
@@ -0,0 +1,11 @@
+﻿using System;
+
+namespace Umbraco.Core.Exceptions
+{
+    class ResolveUnregisteredDependencyException : ApplicationException
```

Comment:
If we are going to have a custom exception, we'll need to follow the best practices, see https://github.com/umbraco/Umbraco-CMS/issues/5947 and https://github.com/umbraco/Umbraco-CMS/pull/6599

That said, we already have LightInjectException, perhaps we can re-use that?

Reply:
Reusing sounds good, will update PR